src/controllers/searchController.ts

```diff
@@ -1,14 +1,57 @@
 import { Request, Response, NextFunction } from 'express';
 import { getStallsInclude, fmtStallsResp } from './stallController';
 import { Stall } from '../models';
+import { generatePaginationWithQueries } from '../utils/paginationUtil';
 
 async function searchStalls(req: Request, res: Response, next: NextFunction) {
   try {
     const rawQuery: string | undefined = req.params.query?.trim();
 
+    const categoryFilter = req.query.category;
+    const categoryFilterCondition = categoryFilter
+      ? `AND id IN (
+            SELECT "stallId"
+            FROM "CategoryStalls"
+            WHERE "categoryId" in (${categoryFilter!
+              .split(',')
+              .map((id: string) => `'${id}'`)
+              .join(',')})
+          )`
+      : '';
+
+    const regionFilter = req.query.region;
+    const regionFilterCondition = regionFilter
+      ? `AND "hawkerCentreId" IN (
+            SELECT id
+            FROM "HawkerCentres"
+            WHERE "regionId" in (${regionFilter!
+              .split(',')
+              .map((id: string) => `'${id}'`)
+              .join(',')})
+          )`
+      : '';
+
+    const limit = +req.query.limit!;
+    const page = +req.query.page!;
+    const offset = (page - 1) * limit;
+
     if (!rawQuery) {
-      const stalls = await Stall.findAll({ include: getStallsInclude() });
-      res.status(200).json(await fmtStallsResp(stalls));
+      const stalls = await Stall.findAndCountAll({
+        order: [['id', 'ASC']],
+        include: getStallsInclude(),
+        limit: limit,
+        offset: offset,
+        distinct: true,
+      });
+      stalls.rows = await fmtStallsResp(stalls.rows);
+      stalls.pagination = generatePaginationWithQueries(
```

Comment:
This one can try using setDataValue("pagination",  generatePaginationWithQueries(.....);

Reply:
The setDataValue doesn't exist for stalls as stalls is not a model.